Softeq.XToolkit.Remote/Client/HttpMessageHandlerBuilder.cs

```diff
@@ -18,99 +18,110 @@ namespace Softeq.XToolkit.Remote.Client
     public abstract class HttpMessageHandlerBuilder
     {
         /// <summary>
-        ///     Gets or sets the primary <see cref="T:System.Net.Http.HttpMessageHandler"/>.
+        ///     Gets or sets the primary <see cref="T:System.Net.Http.HttpMessageHandler"/> that
+        ///     represents the destination of the HTTP message channel.
         /// </summary>
-        public virtual HttpMessageHandler? PrimaryHandler { get; set; }
+        public HttpMessageHandler PrimaryHandler { get; set; } = null!;
 
         /// <summary>
         ///     Gets a list of additional <see cref="T:System.Net.Http.DelegatingHandler"/> instances used to configure an
         ///     <see cref="T:System.Net.Http.HttpClient"/> pipeline.
         /// </summary>
-        public abstract IList<DelegatingHandler> AdditionalHandlers { get; }
+        public IList<DelegatingHandler> AdditionalHandlers { get; } = new List<DelegatingHandler>();
 
         /// <summary>
-        ///     Adds a <see cref="T:System.Net.Http.HttpMessageHandler"/> to the chain of handlers.
+        ///     Adds a <see cref="T:System.Net.Http.HttpMessageHandler"/> to the end of the handlers list.
         /// </summary>
         /// <param name="handler">Handler.</param>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///     When the <paramref name="handler"/> parameter is null.
+        /// </exception>
         /// <returns>Current builder.</returns>
         public HttpMessageHandlerBuilder AddHandler(DelegatingHandler handler)
         {
+            if (handler == null)
+            {
+                throw new ArgumentNullException(nameof(handler));
+            }
+
             AdditionalHandlers.Add(handler);
 
             return this;
         }
 
         /// <summary>
-        ///     Creates an <see cref="T:System.Net.Http.HttpMessageHandler"/>. Adds <see cref="T:System.Net.Http.DelegatingHandler"/> as the last link of the chain.
+        ///     Creates an <see cref="T:System.Net.Http.HttpMessageHandler"/>.
+        ///     Adds <see cref="T:System.Net.Http.DelegatingHandler"/> as the last link of the chain.
+        ///
```

Comment:
If you want to add an empty paragraph between these two sentences, you should use `<para/>` tag instead of adding a new line directly here.

Reply:
Awesome!